tests/speculation/test_prefetch_status_updated.py

```diff
@@ -0,0 +1,154 @@
+#  Copyright 2025 Google LLC.
+#  Copyright (c) Microsoft Corporation.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import pytest
+from test_helpers import (goto_url, read_JSON_message, send_JSON_command,
+                          subscribe)
+
+from . import wait_for_prefetch_event_with_timeout
+
+
+@pytest.mark.asyncio
+async def test_speculation_subscribe_to_all_events(websocket, context_id):
+    """Test subscribing to all speculation events."""
+    subscription_response = await subscribe(websocket, ["speculation"])
+    assert "subscription" in subscription_response
+    assert isinstance(subscription_response["subscription"], str)

```

Comment:
Please assert that the actual events are emitted, not the response of the `subscribe` command.